packages/hash/frontend/src/blocks/page/BlockContextMenu/LoadEntityMenuContent.tsx

```diff
@@ -0,0 +1,169 @@
+import { faAsterisk, faSearch } from "@fortawesome/free-solid-svg-icons";
+import { isBlockEntity } from "@hashintel/hash-shared/entityStore";
+import {
+  addEntityStoreAction,
+  entityStorePluginStateFromTransaction,
+} from "@hashintel/hash-shared/entityStorePlugin";
+import {
+  Box,
+  InputAdornment,
+  ListItemIcon,
+  ListItemText,
+  MenuItem,
+  MenuList,
+} from "@mui/material";
+import { PopupState } from "material-ui-popup-state/core";
+import { useCallback, useEffect, useRef, VFC } from "react";
+import { BlockEntity } from "@hashintel/hash-shared/entity";
+import { useBlockView } from "../BlockViewContext";
+import { FontAwesomeIcon } from "../../../shared/icons";
+import { useRouteAccountInfo } from "../../../shared/routing";
+import { LoadingSpinner, TextField } from "../../../shared/ui";
+import { useAccountEntities } from "../../../components/hooks/useAccountEntities";
+
+type LoadEntityMenuContentProps = {
+  entityId: string | null;
+  popupState?: PopupState;
+};
+
+export const LoadEntityMenuContent: VFC<LoadEntityMenuContentProps> = ({
+  entityId,
+  popupState,
+}) => {
+  const { accountId } = useRouteAccountInfo();
+  const { data: entities, fetchEntities, loading } = useAccountEntities();
+  const blockView = useBlockView();
+  const searchInputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (popupState?.isOpen) {
+      searchInputRef.current?.focus();
+    }
+  }, [popupState]);
+
+  useEffect(() => {
+    const entityStore = entityStorePluginStateFromTransaction(
+      blockView.view.state.tr,
+      blockView.view.state,
+    ).store;
+    const blockData = entityId ? entityStore.saved[entityId] : null;
+
+    if (isBlockEntity(blockData) && accountId) {
+      void fetchEntities(accountId, {
+        componentId: blockData.properties.componentId,
+      });
+    }
+  }, [blockView, entityId, accountId, fetchEntities]);
+
+  const handleClick = useCallback(
+    (targetData: BlockEntity) => {
+      let currentEntityStore = entityStorePluginStateFromTransaction(
+        blockView.view.state.tr,
+        blockView.view.state,
+      ).store;
+      const blockData = entityId ? currentEntityStore.saved[entityId] : null;
+
+      const targetEntity = isBlockEntity(targetData)
+        ? targetData.properties.entity
+        : null;
+      const blockEntity = isBlockEntity(blockData)
+        ? blockData.properties.entity
+        : null;
+
+      if (
+        !targetEntity ||
+        !blockData ||
+        !blockEntity ||
+        targetEntity.entityId === blockEntity.entityId
+      ) {
+        return;
+      }
+
+      const tr = blockView.view.state.tr;
+
+      addEntityStoreAction(blockView.view.state, tr, {
+        type: "updateBlockEntityProperties",
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore -- @todo fix typings
+        payload: {
+          targetEntity,
```

Comment:
I think this is because the payload is setup as `UnknownEntity` instead of `EntityStoreType`

Reply:
Thank you